messages/models.tsp

```diff
@@ -3,27 +3,49 @@
  * Edits made directly to this file will be lost.
  */
 
+import "../assistants";
+
 using TypeSpec.OpenAPI;
 
 namespace OpenAI;
 
+// Tool generated type. Extracts CreateMessageRequest.attachments
+model CreateMessageRequestAttachments
```

Comment:
`file_id` and `tools` are required

Reply:
Interesting, not currently caught because the `required` attribute is on the parent level of the schema, not the schema under `items` where the named properties actually are. I don't know if that's malformed or not in the source material (it seems like it *should* be incorrect, as you'd lose a descriptive layer), but it's comparatively safe enough to add logic for it.

```
        attachments:
          type: array
          items:
            type: object
            properties:
              file_id:
                type: string
                description: The ID of the file to attach to the message.
              tools:
                description: The tools to add this file to.
                type: array
                items:
                  oneOf:
                    - $ref: "#/components/schemas/AssistantToolsCode"
                    - $ref: "#/components/schemas/AssistantToolsFileSearch"
                  x-oaiExpandable: true
          description: A list of files attached to the message, and the tools they should be added to.
          required:
            - file_id
            - tools
          nullable: true
```

Reply:
It looks like we re-emit it the same way, so seems like it's proper OpenAPI .yaml. I guess because we're constrained to `type: array` it works. But it's fixed!